concordium-consensus/src/Concordium/KonsensusV1/TreeState/Implementation.hs

```diff
@@ -343,6 +356,51 @@ mkInitialSkovData rp genMeta genState _currentTimeout _skovEpochBakers =
         _currentQuorumMessages = emptyQuorumMessages
     in  SkovData{..}
 
+-- |Create an initial 'SkovData pv'
+-- This constructs a 'SkovData pv' from a genesis block
+-- which is suitable to grow the tree from.
+mkInitialSkovData ::
+    -- |The 'RuntimeParameters'
+    RuntimeParameters ->
+    -- |Genesis metadata. State hash should match the hash of the state.
+    GenesisMetadata ->
+    -- |Genesis statep
+    PBS.HashedPersistentBlockState pv ->
+    -- |The base timeout
+    Duration ->
+    -- |Bakers at the genesis block
+    EpochBakers ->
+    -- |The initial 'SkovData'
+    SkovData pv
+mkInitialSkovData rp genMeta genState _currentTimeout _skovEpochBakers = mkInitialSkovData' rp genMeta genState _currentTimeout _skovEpochBakers Nothing
+
+-- |Create an initial 'SkovData pv'.
+-- This constructs a 'SkovData pv' from a genesis block and
+-- carries over the account nonces and sequence numbers from the supplied
+-- transaction table.
+mkInitialSkovDataWithTT ::
+    -- |The 'RuntimeParameters'
+    RuntimeParameters ->
+    -- |Genesis metadata. State hash should match the hash of the state.
+    GenesisMetadata ->
+    -- |Genesis statep
+    PBS.HashedPersistentBlockState pv ->
+    -- |The base timeout
+    Duration ->
+    -- |Bakers at the genesis block
+    EpochBakers ->
+    -- |The transaction table to migrate from.
+    -- This carries over the account nonces and sequence numbers to the
+    -- new transaction table.
+    TransactionTable ->
+    -- |The initial 'SkovData'
+    SkovData pv
+mkInitialSkovDataWithTT rp genMeta genState _currentTimeout _skovEpochBakers tt =
+    let migratedTT = emptyTransactionTableWithSequenceNumbers migratedAccountNonces migratedUpdateSequenceNumbers
+        migratedAccountNonces = map (\(addr, anft) -> (aaeAddress addr, anft ^. anftNextNonce)) (HM.toList $ tt ^. ttNonFinalizedTransactions)
+        migratedUpdateSequenceNumbers = Map.fromList $ map (\(utype, nfcu) -> (utype, nfcu ^. nfcuNextSequenceNumber)) (Map.toList $ tt ^. ttNonFinalizedChainUpdates)
```

Comment:
I don't think this is correct. When this function is used you pass in the existing transaction table, with potentially pending transactions.

And then you only extract these sequence numbers but don't retain the transaction themselves. So unless I'm missing something this won't really work the way it's meant to, transactions will be dropped.

Is there a reason for doing it in a different way than we already do for existing protocol versions, where we just copy over the transaction table?